rosbag2_examples/rosbag2_examples_cpp/src/simple_bag_player.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright 2024 Open Source Robotics Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <iostream>
+#include <memory>
+#include <string>
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp/serialization.hpp"
+#include "rosbag2_cpp/reader.hpp"
+#include "std_msgs/msg/string.hpp"
+
+using namespace std::chrono_literals;
+
+class SimpleBagPlayer : public rclcpp::Node
+{
+public:
+  explicit SimpleBagPlayer(const std::string & bag_filename)
+  : Node("playback_node")
+  {
+    publisher_ = this->create_publisher<std_msgs::msg::String>("chatter", 10);
+
+    timer_ = this->create_wall_timer(
+      100ms,
+      [this]() {return this->timer_callback();}
```

Comment:
So this simple bag player does not care the timestamp in the bag file at all.

if user publishes `chatter` topic with 100Hz (10 msec) and record it with `simple_bag_recorder`, this will actually replay the messages with 1/10 frequency against the bag file timestamps.

i think this is no problem at all as example, but i think user needs to be aware of this situation. how about the adding comment somewhere here to explain this?

Reply:
Resolved